sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/SqlMetadata.java

```diff
@@ -262,4 +266,50 @@ public void addOutputType(final String outputName, final String colType) {
     public String getOutputType(final String outputName, final String colType) {
         return aggregateTypeMap.getOrDefault(outputName, colType);
     }
+
+    public String getType(final String column) throws SQLException {
+        final List<GremlinTableBase> gremlinTableBases = new ArrayList<>();
+        for (final String table : getColumnOutputListMap().keySet()) {
+            gremlinTableBases.add(getGremlinTable(table));
+        }
+        if (aggregateTypeExists(column)) {
+            return getOutputType(column, "string");
+        }
+        String renamedColumn = getRenamedColumn(column);
+        if (!aggregateTypeExists(renamedColumn)) {
+            // Sometimes columns are double renamed.
+            renamedColumn = getRenamedColumn(renamedColumn);
```

Comment:
If columns are double renamed, shall it be before the if condition?

Reply:
If aggregateTypeExists passes then we already know the output type so no point.